core/src/main/java/com/scalar/db/api/builder/PutBuilder.java

```diff
@@ -0,0 +1,416 @@
+    public Buildable intValue(String columnName, @Nullable Integer value) {
+      if (value != null) {
+        return intValue(columnName, value.intValue());
+      }
+      columns.put(columnName, IntColumn.ofNull(columnName));
+      return this;
+    }
+
+    @Override
+    public Buildable bigIntValue(String columnName, long value) {
+      columns.put(columnName, BigIntColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable bigIntValue(String columnName, @Nullable Long value) {
+      if (value != null) {
+        return bigIntValue(columnName, value.longValue());
+      }
+      columns.put(columnName, BigIntColumn.ofNull(columnName));
+      return this;
+    }
+
+    @Override
+    public Buildable floatValue(String columnName, float value) {
+      columns.put(columnName, FloatColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable floatValue(String columnName, @Nullable Float value) {
+      if (value != null) {
+        return floatValue(columnName, value.floatValue());
+      }
+      columns.put(columnName, FloatColumn.ofNull(columnName));
+      return this;
+    }
+
+    @Override
+    public Buildable doubleValue(String columnName, double value) {
+      columns.put(columnName, DoubleColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable doubleValue(String columnName, @Nullable Double value) {
+      if (value != null) {
+        return doubleValue(columnName, value.doubleValue());
+      }
+      columns.put(columnName, DoubleColumn.ofNull(columnName));
+      return this;
+    }
+
+    @Override
+    public Buildable textValue(String columnName, @Nullable String value) {
+      columns.put(columnName, TextColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable blobValue(String columnName, @Nullable byte[] value) {
+      columns.put(columnName, BlobColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable blobValue(String columnName, @Nullable ByteBuffer value) {
+      columns.put(columnName, BlobColumn.of(columnName, value));
+      return this;
+    }
+
+    @Override
+    public Buildable value(Column<?> column) {
+      columns.put(column.getName(), column);
+      return this;
+    }
+
+    @Override
+    public Put build() {
+      Put put = new Put(partitionKey, clusteringKey);
+      put.forNamespace(namespaceName).forTable(tableName);
+      columns.values().forEach(put::withValue);
+      if (consistency != null) {
+        put.withConsistency(consistency);
+      }
+      if (condition != null) {
+        put.withCondition(condition);
+      }
+
+      return put;
+    }
+
+    @Override
+    public Buildable consistency(com.scalar.db.api.Consistency consistency) {
+      checkNotNull(consistency);
+      this.consistency = consistency;
+      return this;
+    }
+  }
+
+  public static class BuildableFromExisting
```

Comment:
Just a question.
Can we possibly unify Buildable and BuildableFromExisting?
Essentially, the difference between them is whether or not they have default values, so I'm wondering.
(Having clearValue(s) methods is the reason?)

Reply:
Buildable and BuildableFromExisting can't be completely unified because the BuildableFromExisting need to expose methods that Buildable don't such as namespace(), table(), partitionKey(), clearValues(), etc.
Though, I think I can make BuildableFromExisting extends Buildable. In BuildableFromExisting, all inherited methods will need to be overridden because they need to return the BuildableFromExisting type and not the super() class type.
It should save a bit of duplicated code.

Reply:
Thank you!
Then, composition might be more appropriate than inheritance.

Reply:
Thank you for the suggestion. Composition is also doable, but ultimately, I think making **BuildableFromExisting** inherit **Buildable** will yield a better code factorization and a design that is more simple. 
In my opinion, each builder method logic is quite simple (more or less a setter), so creating a concrete class for each kind of builder attribute (NamespaceName, TableName, etc) might be too lengthy.